nav2_map_server/src/map_saver.cpp

```diff
@@ -30,221 +31,173 @@
-      throw std::runtime_error("Threshold_free must be smaller than threshold_occupied");
-    }
+  RCLCPP_INFO(get_logger(), "Creating");
 
-    std::string mode_str = declare_parameter("map_mode", "trinary");
-    try {
-      map_mode = map_mode_from_string(mode_str);
-    } catch (std::invalid_argument &) {
-      map_mode = MapMode::Trinary;
-      RCLCPP_WARN(
-        get_logger(), "Map mode parameter not recognized: '%s', using default value (trinary)",
-        mode_str.c_str());
-    }
+  save_map_timeout_ = std::make_shared<rclcpp::Duration>(
+    std::chrono::milliseconds(declare_parameter("save_map_timeout", 2000)));
 
-    image_format = declare_parameter("image_format", map_mode == MapMode::Scale ? "png" : "pgm");
-    std::transform(
-      image_format.begin(), image_format.end(), image_format.begin(),
-      [](unsigned char c) {return std::tolower(c);});
-    const std::vector<std::string> BLESSED_FORMATS{"bmp", "pgm", "png"};
-    if (
-      std::find(BLESSED_FORMATS.begin(), BLESSED_FORMATS.end(), image_format) ==
-      BLESSED_FORMATS.end())
-    {
-      std::stringstream ss;
-      bool first = true;
-      for (auto & format_name : BLESSED_FORMATS) {
-        if (!first) {
-          ss << ", ";
+  free_thresh_default_ = declare_parameter("free_thresh_default", 25),
+  occupied_thresh_default_ = declare_parameter("occupied_thresh_default", 65);
+
+  map_listener_ = rclcpp::Node::make_shared("map_listener");
+}
+
+MapSaver::~MapSaver()
+{
+  RCLCPP_INFO(get_logger(), "Destroying");
+}
+
+nav2_util::CallbackReturn
+MapSaver::on_configure(const rclcpp_lifecycle::State & /*state*/)
+{
+  RCLCPP_INFO(get_logger(), "Configuring");
+  // Create SaveMap service callback handle
+  auto save_map_callback= [this](
+    const std::shared_ptr<rmw_request_id_t>/*request_header*/,
+    const std::shared_ptr<nav2_msgs::srv::SaveMap::Request> request,
+    std::shared_ptr<nav2_msgs::srv::SaveMap::Response> response) -> void {
+        // Set input arguments and call saveMapTopicToFile()
+        SaveParameters save_parameters;
+        save_parameters.map_file_name = request->map_url;
+        save_parameters.image_format = request->image_format;
+        save_parameters.free_thresh = request->free_thresh;
+        save_parameters.occupied_thresh = request->occupied_thresh;
+        try {
+          save_parameters.mode = map_mode_from_string(request->map_mode);
+        } catch (std::invalid_argument &) {
+          save_parameters.mode = MapMode::Trinary;
+          RCLCPP_WARN(
+            get_logger(), "Map mode parameter not recognized: '%s', using default value (trinary)",
+            request->map_mode.c_str());
         }
-        ss << "'" << format_name << "'";
-        first = false;
-      }
+
+        std::string map_topic = request->map_topic;
+
+	response->result = saveMapTopicToFile(map_topic, save_parameters);
+    };
+
+  // Make name prefix for services
+  std::string service_prefix = get_name() + std::string("/");
```

Comment:
```suggestion
  const std::string service_prefix = get_name() + std::string("/");
```

Reply:
Done: https://github.com/ros-planning/navigation2/commit/89ece9672dc847161607acc658b63fe2bafb90a7#diff-99885e6562ca50edd8cc80db6362d317R49 and https://github.com/ros-planning/navigation2/commit/89ece9672dc847161607acc658b63fe2bafb90a7#diff-a42e5d09c9ddddb03729320e0be84f12R103